src/i18n/i18n-loader.js

```diff
@@ -46,9 +47,20 @@ LANGUAGES.registerLocale(require('@cospired/i18n-iso-languages/langs/en.json'));
 LANGUAGES.registerLocale(require('@cospired/i18n-iso-languages/langs/es.json'));
 LANGUAGES.registerLocale(require('@cospired/i18n-iso-languages/langs/fr.json'));
 
-// TODO (ARCH-563): this should ultimately come from the user's settings, but for now, set your
-// browser language to the language you want to see
-const getLocale = (localeStr = window.navigator.language) => localeStr.substr(0, 2);
+const cookies = new Cookies();
+
+// Get the locale by setting priority
+const getLocale = (localeStr) => {
+  // 1. Explicit application request
+  if (localeStr) return localeStr;
+
+  // 2. User setting in cookie
+  const cookieLanguagePreference = cookies.get(process.env.LANGUAGE_PREFERENCE_COOKIE_NAME);
+  if (cookieLanguagePreference) return cookieLanguagePreference;
```

Comment:
We should probably check if we actually offer that language... maybe even for #1 above.